docs/pyproject.md

```diff
@@ -13,9 +13,17 @@ menu:
 
 The `tool.poetry` section of the `pyproject.toml` file is composed of multiple sections.
 
+## mode
+
+The mode of the project, either `"package"` (default) or `"non-package"`. **Optional**
```

Comment:
It might be nice to link to the section in `basic-usage.md` from here, or to some other explanation of what this means.

Alternatively, you could shorten the description of non-package mode there and move it here.